src/main/java/com/example/employee/exceptionHandler/EmployeeExceptionHandler.java

```diff
@@ -0,0 +1,29 @@
+package com.example.employee.exceptionHandler;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestControllerAdvice
+public class EmployeeExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value = {IllegalArgumentException.class})
```

Comment:
Let's handle our new custom exception here instead of all `IllegalArgumentException`.